### PR TITLE
Initial entry for QAT compression/decompression support

### PR DESCRIPTION
**Pull request by ivakegg**

### CODE REVIEW

File: core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/Compression.java [hunk 1]
```diff
@@ -429,6 +432,115 @@ public boolean isSupported() {
...
+      @Override
+      public InputStream createDecompressionStream(InputStream downStream, Decompressor decompressor, int downStreamBufferSize) throws IOException {
+        if (!isSupported()) {
+          throw new IOException("QAT codec class not specified. Did you forget to set property " + CONF_QAT_CLASS + "?");
+        }
+
+        CompressionCodec decomCodec = qatCodec;
+        // if we're not using the same buffer size, we'll pull the codec from the loading cache
+        if (DEFAULT_BUFFER_SIZE != downStreamBufferSize) {
+          Entry<Algorithm,Integer> sizeOpt = Maps.immutableEntry(SNAPPY, downStreamBufferSize);
```
Comment (dlmarion) [inline core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/Compression.java]: I think the key in the map should be QAT, no?

Comment (ivakegg) [inline core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/Compression.java]: missed one, good catch

**Comment by dlmarion** [comment]
LGTM

**Comment by dlmarion** [comment]
LGTM

**Pull request closed without merging**